providers/CENNZApiProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import { Api } from "@cennznet/api";
+import { createContext, useState, PropsWithChildren, useContext } from "react";
+
+type CENNZApiContextType = {
+  api: Api;
+  updateApi: Function;
+};
+
+const CENNZApiContext = createContext<CENNZApiContextType>(null);
+
+export default function CENNZApiProvider({ children }: PropsWithChildren<{}>) {
+  const [api, setApi] = useState<Api>(null);
+
+  const updateApi = (endpoint) => {
```

Comment:
This is optional.
I would wrap this in a `useCallback` to keep it stable across render cycles

Reply:
done